File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        mp = {}

        for i in range(len(nums)):
            a = target - nums[i]

            if a in mp:
                return [mp[a] , i]
        
            mp[nums[i]] = i

        return [-1 ,-1]

File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int n = nums.size();
        unordered_map<int , int> mp;
        vector<int> ans;

        for(int i=0; i<n; i++){

            int num = target - nums[i];
            
            if(mp.find(num) != mp.end()){
                ans.push_back(mp[num]);
                ans.push_back(i);
                return ans;
            }
            mp[nums[i]] = i;
        }
       return ans;
    }
};